src/lib/types/restrictions.ts

```diff
@@ -77,36 +99,25 @@ const fixedOutputMetadataValueType = z.object({
   metadataValueType: metadataValueType,
 })
 
-export const userRestrictionValidation = z.object({
-  None: z.array(none).optional(),
-  SenderOwnsAllInputs: z.array(senderOwnsAllInputs).optional(),
-  SenderHasInputRole: z.array(senderHasInputRole).optional(),
-  SenderHasOutputRole: z.array(senderHasOutputRole).optional(),
-  OutputHasRole: z.array(outputHasRole).optional(),
-  MatchInputOutputRole: z.array(matchInputOutputRole).optional(),
-  MatchInputOutputMetadataValue: z.array(matchInputOutputMetadataValue).optional(),
-  FixedNumberOfInputs: z.array(fixedNumberOfInputs).optional(),
-  FixedNumberOfOutputs: z.array(fixedNumberOfOutputs).optional(),
-  FixedInputMetadataValue: z.array(fixedInputMetadataValue).optional(),
-  FixedOutputMetadataValue: z.array(fixedOutputMetadataValue).optional(),
-  FixedOutputMetadataValueType: z.array(fixedOutputMetadataValueType).optional(),
+export const stepValidation = z.object({
```

Comment:
Instead of doing this as an object with a bunch of optional fields (where its only correct to have exactly one present) could this be represented by a union type? It wouldn't have to change the JSON but would make the types more accurate

Reply:
yes!

Reply:
This is something I will address with the following PR, if no objections.